### DISCUSSION TITLE
who answers "tft.width()/tft.height()

### DISCUSSION THREAD

**Discussion by MHz000** [Bodmer/TFT_eSPI#2847] [Q&A - General]
1) Is a driver internal value returned as a response or does the driver query the graphics chip and return this value?

2) Is there a way to query the type and subtype (color) directly from the chip

**Comment by Bodmer** [comment]
To answer:
1. No and no, the value comes from either the default set in the library, or the width and height set in the setup file.
2. Possibly, some display boards allow read access to registers. The register values to read varies from display to display, so consult the drivvers datasheet for more information.
